File: MobileApp/SmartGreenhouseApp/app/src/main/java/com/group22/smartgreenhouse/data/model/ElectronicCard.kt
package com.group22.smartgreenhouse.data.model

// Device.kt
data class ElectronicCard(
    val id: Int,
    val productName: String?,
    val greenHouseId: Int?,
    val status: String, // "Available", "Unavailable", "Error"
    val lastDataTime: String?,
    val temperature: String,
    val humidity: String,
    val errorState: String?,
    val userId: String,
    val createdBy: String,
    val created: String,
    val lastModifiedBy: String?,
    val lastModified: String?
)

data class DeviceListResponse(
    val cards: List<ElectronicCard>,
    val totalCount: Int,
    val pageSize: Int,
    val currentPage: Int,
    val totalPages: Int
)

File: Backend/CleanArchitecture/CleanArchitecture.Application/Entities/ElectronicCard.cs
namespace CleanArchitecture.Core.Entities;

public class ElectronicCard : AuditableBaseEntity
{
    public string ProductName { get; set; }
    public int? GreenHouseId { get; set; }
    public string Status { get; set; }
    public string LastDataTime { get; set; }
    public string Temperature { get; set; }
    public string Humidity { get; set; }
    public string ErrorState { get; set; }
    public string UserId { get; set; }
}